app/models/reports/repository.rb

```diff
@@ -0,0 +1,100 @@
+module Reports
+  class Repository
```

Comment:
> * Have numbers be first class entities (Think `Reports::ControlRate`, `Reports::Registrations`). These can be independently cached and the controller can pull numbers straight from them. I think this might make it easier to trace numbers from view to the query (which is a bit hard right now). This has some overlap with dismantling `result.rb`.

Yup, I think something like this definitely makes sense.